ARROW-8425: [Rust] [Parquet] Correct temporal IO

nevi-me opened the pull request:
Contains the following:
- Fixes the date64 writer by writing the array as a timestamp milli
- Adds support for reading and writing interval types

> Apache Spark doesn't read intervals (as they're fixed len binary)
> It looks like the CPP impl at v2.0.0 just reads back the binary data.
The binary data read is however correct.

The result is that we can now correctly read and write the temporal types.
The Duration type remains unsupported.

nevi-me commented:
CC @carols10cents @houqp

jorgecarleitao reviewed (commented):
I did not see any changes in the tests, which I would expect for a change in semantics. Aren't we not testing this yet, or how can I verify that it is now correct?

nevi-me commented:
> I did not see any changes in the tests, which I would expect for a change in semantics. Aren't we not testing this yet, or how can I verify that it is now correct?

There were existing tests, but they were ignored. So I removed the `#[ignore]` so they run.

I was writing date64 as timestampmilli, but this wasn't per the spec, so I fixed that now. I also added an interval test to increase test coverage.

jorgecarleitao reviewed (approved):
LGTM.
